### DISCUSSION TITLE
Contradiction in weight calculated by `.predict()` and manual calculation

### DISCUSSION THREAD

**Discussion by shivam221098** [moj-analytical-services/splink#2034] [Q&A] [answered]
I'm trying to run a deduplication job. 

When I do `linker.predict()`, the weights that this method calculates under the `match_weight` column, are negative, but when I see gamma values from the table returned by the `.predict()` method, and sum up partial match weights returned from `.match_weight_chart()` corresponding to each gamma weight and prior weight, I get a high positive weight above my threshold, Can't say how this `.predict()` method calculates this match weights.

I'm also not able to link all rows that are matching i.e. all column values are the same.
![Screenshot 2024-03-06 at 8 25 16 PM](https://github.com/moj-analytical-services/splink/assets/56783600/35a00650-7cba-444d-b443-591a47e63d14)

```
gamma_lastname = 1, match_weight = 11.0127
gamma_firstname = 1, match_weight = 9.1043
gamma_secondname = 1, match_weight = 1.5119
gamma_email = -1, match_weight = 0 (as email_l is null)
gamma_{hidden_col} = -1, match_weight = 0 (as left is null)
gamma_custom_{hidden_col1} = 5, match_weight = 5.7269
prior_weight = -14.4030
```

Does anyone have any idea?

**Comment by RobinL** [comment]
Take a look at the waterfall chart for the specific record pair, it should describe the full calculation:
https://moj-analytical-services.github.io/splink/charts/waterfall_chart.html

**Comment by shivam221098** [comment, marked as answer]
I appreciate the time if anyone looked at it. 
I just saw my mistake, I marked `term_frequency_adjustments` to `True` and this was decreasing the weight when I get exact matches.

Silly mistake, I saw after trying it for the whole day. 

Thanks.
